### DISCUSSION TITLE
Feature MempoiSheet "withHeaderTranslation"

### DISCUSSION THREAD

**Discussion by bdzzaid** [firegloves/MemPOI#22] [Ideas]
Hi there, it's me again :-) 

I'm still migrating from Jasper to MemPOI. I need now to translate the header name. 

As it's mentionned in the specification here : 
```markdown
# Column headers

Column headers are generated taking export query column names. 
If you want to choose column headers you need to speficy them with AS clause. 
For example:

SELECT id, name AS first_name FROM Foo

will result in a sheet with 2 columns: id and first_name (containing db's name column data)
```
This limitation is too bad for a daily usage of this API in production. 

May I suggest to provide in the MempoiSheet a simple Map<String,String> for columns headers translation during the process.
```java
Map<String, String> map = new LinkedHashMap<>();
map.put("firstName", "Prénom");
map.put("lastName", "Nom");
map.put("birthday", "Date de naissance");
MempoiSheet exportSheet = MempoiSheetBuilder.aMempoiSheet()
      .withHeaderTranslation(map)
```
And instead of generating this Excel
![image](https://user-images.githubusercontent.com/11232571/116413384-dbd88580-a837-11eb-82d7-9d1bc6e55215.png)

I will have the translated result by mapping the ResultSet column header key to the map translated key 

![image](https://user-images.githubusercontent.com/11232571/116415004-4b9b4000-a839-11eb-92d5-1884496d3928.png)

This is a mandatory requirement to go live in production. Feel free to tell me if I can contribute to develop this feature.

Zaïd

**Comment by bdzzaid** [comment]
Hi there !  I finally forked the MemPOI project to implement this feature. 

@firegloves @mirkoPan I will add the String attribute "columnDisplayName" to MempoiColumnConfigBuilder and if present, I'll export the display name of the column instead of the SQL resultset column name : 
![image](https://user-images.githubusercontent.com/11232571/116530048-dfbee300-a8dd-11eb-8234-1db80fa23558.png)

Are you OK with this implementation ?

**Comment by firegloves** [comment]
@bdzzaid due to the fact that you need this feature as soon as possible and that we don't have the required time to follow the development as we would like, I'm publishing a snapshot with your recent developments, in order to unblock you with your job.

I'll ping you when the deployment will be ready, please let us know if you find some problems

**Reply by bdzzaid** [reply]
@firegloves Thanks :-)

**Reply by firegloves** [reply]
@bdzzaid you are welcome :)

it's finally reachable, using gradle:

```
repositories {
    mavenCentral()

    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
}

dependencies {
    compile group: 'it.firegloves', name: 'mempoi', version: '1.6.0-SNAPSHOT'
}
```
